middle_end/closure/closure.ml

```diff
@@ -861,7 +861,7 @@ let close_var env id =
 let rec close ({ backend; fenv; cenv ; mutable_vars } as env) lam =
   let module B = (val backend : Backend_intf.S) in
   match lam with
-  | Lvar id ->
+  | Lvar id | Lmutvar id ->
```

Comment:
Here, if I read the code correctly, `Lmutvar id` will never have a value optimization in the store, so we could return `Uvar id` directly.

(In the future if we had a mutability flag on `Uvar` as well we could get rid of `mutable_variables`; but this would be a more invasive change to consider later.)